### PR TITLE
Add neo4j health check

### PR DESCRIPTION
**Pull request by a-z-hub**
**What this PR does / why we need it**:
This PR add health check for [neo4j](https://neo4j.com/)

**Which issue(s) this PR fixes**:
It is new project and implementation.

Please reference the issue this PR will close: #509 

**Special notes for your reviewer**:
user nuget package [Neo4j.Driver](https://www.nuget.org/packages/Neo4j.Driver/)

**Does this PR introduce a user-facing change?**:
Yes. New extension mentod `.AddNeo4j(...)` for `IHealthChecksBuilder` and create new nuget package `AspNetCore.HealthChecks.Neo4j`

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [x] Created/updated tests
- [x] Unit tests passing
- [x] End-to-end tests passing
- [x] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

File: src/HealthChecks.Neo4j/HealthChecks.Neo4j.csproj [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  <PropertyGroup>
+    <TargetFramework>$(NetStandardTargetVersion)</TargetFramework>
+    <PackageLicenseUrl>$(PackageLicenseUrl)</PackageLicenseUrl>
+    <PackageProjectUrl>$(PackageProjectUrl)</PackageProjectUrl>
+    <Description>HealthChecks.Neo4j is the health check package for Neo4j.</Description>
+    <PackageTags>HealthCheck;Health;Neo4j</PackageTags>
+    <Version>$(HealthCheckNeo4j)</Version>
+    <RepositoryUrl>$(RepositoryUrl)</RepositoryUrl>
+    <Company>$(Company)</Company>
+    <Authors>$(Authors)</Authors>
```
Comment (sungam3r) [inline src/HealthChecks.Neo4j/HealthChecks.Neo4j.csproj:11]: Please remove such self-assignments,

File: src/HealthChecks.Neo4j/Neo4jHealthCheck.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                using (var driver = GraphDatabase.Driver(_options.Uri, AuthTokens.Basic(_options.UserName, _options.Password)))
+                {
+                    var _ = driver.AsyncSession();
+                }
+				return Task.FromResult(HealthCheckResult.Healthy());
+			}
+            catch (Exception ex)
+            {
+				return Task.FromResult(new HealthCheckResult(context.Registration.FailureStatus, exception: ex));
+			}
```
Comment (sungam3r) [inline src/HealthChecks.Neo4j/Neo4jHealthCheck.cs:30]: indentation is broken

File: test/FunctionalTests/HealthChecks.Neo4j/Neo4jHealthCheckTests.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+               {
+                   services.AddHealthChecks()
+                   .AddNeo4j(_ => options, tags: new[] { "test-neo4j" });
+               })
+               .Configure(app =>
+               {
+                   app.UseHealthChecks("/health", new HealthCheckOptions()
+                   {
+                       
+
```
Comment (sungam3r) [inline test/FunctionalTests/HealthChecks.Neo4j/Neo4jHealthCheckTests.cs:48]:
```suggestion
```

File: src/HealthChecks.Neo4j/HealthChecks.Neo4j.csproj (hunk 1 above)
Comment (ahryshchanka) [reply src/HealthChecks.Neo4j/HealthChecks.Neo4j.csproj:11]: - PackageLicenseUrl
- PackageProjectUrl
- RepositoryUrl
- Company
- Authors
- PublishRepositoryUrl
removed

File: src/HealthChecks.Neo4j/Neo4jHealthCheck.cs (hunk 2 above)
Comment (ahryshchanka) [reply src/HealthChecks.Neo4j/Neo4jHealthCheck.cs:30]: reformated

File: test/FunctionalTests/HealthChecks.Neo4j/Neo4jHealthCheckTests.cs (hunk 3 above)
Comment (ahryshchanka) [reply test/FunctionalTests/HealthChecks.Neo4j/Neo4jHealthCheckTests.cs:48]: fixed

**Comment by ahryshchanka** [comment]
@sungam3r thank you for review :smiley: :+1:

**Comment by unaizorrilla** [comment]
Hi @ahryshchanka 

I try to review this weekend!

**Comment by unaizorrilla** [comment]
Hi @ahryshchanka 

I'm check the health check and I'm not sure if all cases are covered! The functional tests are working on AppVeyor without the SkipOnAppVeyor attribute, how is posible the test be_healthy_when_neo4j_server_is_available works without neo4j running instance?

This code

`_ = driver.AsyncSession();`

Fail or notify failure if the no running instance is listenning or password are invalid??

**Comment by unaizorrilla** [comment]
Hi @ahryshchanka

Are you plan to address my feedback on this PR?

**Comment by ahryshchanka** [comment]
Hi @unaizorrilla 

I did changes with the query `MATCH (n) RETURN count(n) as count`
Approach with code `_ = driver.AsyncSession();` not working for me.

Sorry for waiting

**Comment by ahryshchanka** [comment]
Hi @unaizorrilla 

Can you please help me with build issue investigations?
On my local env all work correctly. Possibly you can attach docker logs?

**Comment by allmywatts** [comment]
Is this still being worked on?

**Comment by a-z-hub** [comment]
@allmywatts @unaizorrilla I did rebase and update all pr. I can't check if it works locally (I have ios only).
Can we run CI/CD pipeline?

**Comment by allmywatts** [comment]
@a-z-hub Hey, could you use an even simpler cypher query like `RETURN 1`? I think you can get an exception with the match query in the enterprise edition of neo4j if the user does not have certain permissions. 😃

**Pull request closed without merging**